### PR TITLE
Move YDocWebSocketHandler to jupyter-server

### LINKED ISSUE #12770: Move YDocWebSocketHandler to Jupyter Server code
### Problem

I think that the [YDocWebSocketHandler](https://github.com/jupyterlab/jupyterlab/blob/055099fa565f3589d8db38588678614fdb286023/jupyterlab/handlers/ydoc_handler.py) is not specific to JupyterLab and deserves to live in [jupyter-server](https://github.com/jupyter-server/jupyter_server) core.
It is a generic websocket server that can talk to any websocket provider for the supported Y documents. JupyterLab is just one of them, but I can imagine e.g. a Python web client that would also have a websocket provider, such as [ypy-websocket](https://github.com/y-crdt/ypy-websocket/blob/39252430d79141ec7900cda8c291996da6aaf340/ypy_websocket/websocket_provider.py), to implement a Jupyter client in the terminal. In that case we wouldn't need to run JupyterLab in the back-end but only Jupyter Server.

### Proposed Solution

Move [ydoc_handler.py](https://github.com/jupyterlab/jupyterlab/blob/055099fa565f3589d8db38588678614fdb286023/jupyterlab/handlers/ydoc_handler.py) to [jupyter-server](https://github.com/jupyter-server/jupyter_server).

### Additional context

This is also needed if we want to implement a [new kernels REST API](https://github.com/jupyter-server/jupyter_server/issues/900) in jupyter-server.

### PR DESCRIPTION
**Pull request by davidbrochart**
## References

Closes #12770.

## Code changes

The `YDocWebSocketHandler` moved to [jupyter_server_ydoc](https://github.com/jupyter-server/jupyter_server_ydoc).

## User-facing changes

None.

## Backwards-incompatible changes

None.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -41,8 +41,6 @@ install_requires =
     jupyter_server>=1.16.0,<2
     notebook_shim>=0.1
     jinja2>=3.0.3
-    ypy-websocket
-    jupyter_ydoc==0.1.10
```
Comment (jtpio) [inline setup.cfg:45]: Probably this will also require bumping the minimum version of `jupyter_server` if / when it lands there?

**Comment by davidbrochart** [comment]
Tests are failing because jupyter_server_ydoc has not been released yet.

**Comment by davidbrochart** [comment]
I think the CI failures are not related.

**Comment by davidbrochart** [comment]
The `YDocWebSocketHandler` moved to [jupyter_server_ydoc](https://github.com/jupyter-server/jupyter_server_ydoc).
This is ready to be merged.

**Comment by davidbrochart** [comment]
OK to merge?

**Review by jtpio — APPROVED** [review]
Thanks!

**Comment by jtpio** [comment]
The `release_test` failure was fixed in https://github.com/jupyterlab/jupyterlab/pull/12799

**Pull request merged**